src/event_manager/manager.rs

```diff
@@ -0,0 +1,416 @@
+use std::sync::{Arc, Mutex};
+
+use crate::epoll::{ControlOperation, EpollEvent, EventSet};
+
+use super::endpoint::EventManagerChannel;
+use super::epoll_context::EpollContext;
+use super::subscribers::Subscribers;
+use super::{
+    ControlOps, Error, EventSubscriber, Events, RemoteEndpoint, Result, SubscriberId, SubscriberOps,
+};
+
+/// Allows event subscribers to be registered, connected to the event loop, and later removed.
+pub struct EventManager<T> {
+    subscribers: Subscribers<T>,
+    epoll_context: EpollContext,
+    ready_events: Vec<EpollEvent>,
```

Comment:
Like the idea to use a statically  allocated buffer for EpollEvemt, actually we have adopted a similar mechanism to avoid allocating/freeing memory on each epoll loop.